.github/workflows/release.yml

```diff
@@ -15,19 +15,22 @@ jobs:
     env:
       DOCKER_REGISTRY: docker.elastic.co
       DOCKER_SECRET: secret/observability-team/ci/docker-registry/prod
-      DOCKER_IMAGE_NAME: apm-attacher
+      DOCKER_IMAGE_NAME: observability/apm-attacher
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: elastic/apm-pipeline-library/.github/actions/docker-login@current
         with:
           registry: ${{ env.DOCKER_REGISTRY }}
           secret: ${{ env.DOCKER_SECRET }}
           url: ${{ secrets.VAULT_ADDR }}
           roleId: ${{ secrets.VAULT_ROLE_ID }}
           secretId: ${{ secrets.VAULT_SECRET_ID }}
-      - name: Package & Publish
+      - name: Docker build, tag, and push
         run: |
-          ./.ci/docker-package-push.sh ${DOCKER_REGISTRY}/observability ${DOCKER_IMAGE_NAME} ${{ github.ref_name }}
```

Comment:
Note for reviewers: The "docker-package-push.sh" script *had* been used by both release.yml and publish.yml. Now that the latter is gone, there wasn't much value in another level of indirection for the build steps. 

I think it *would* be valuable for these steps to live in the top-level Makefile or a top-level scripts/foo.sh, rather than be scoped to just CI stuff, but I didn't want to do that.